qiskit/providers/fake_provider/fake_backend.py

```diff
@@ -344,98 +349,23 @@ def run(self, run_input, **options):
         if pulse_job:  # pulse job
             raise QiskitError("Pulse simulation is currently not supported for V2 fake backends.")
         # circuit job
-        if not _optionals.HAS_AER:
+
+        if _optionals.HAS_AER:
+            from pkg_resources import parse_version
```

Comment:
Terra doesn't have a direct requirement on `pkg_resources` (and this module is effectively deprecated anyway), so we can't use this.